.github/workflows/release-code-freeze.yml

```diff
@@ -257,3 +257,19 @@ jobs:
           WPCOM_RELEASE_WEBHOOK_URL: ${{ secrets.WPCOM_RELEASE_WEBHOOK_URL }}
           ENVIRONMENT: ${{ secrets.ENVIRONMENT }}
 
+  notify-slack-on-failure:
+    name: Notify Slack on Failure
+    runs-on: ${{ ( github.repository == 'woocommerce/woocommerce' && 'blacksmith-2vcpu-ubuntu-2404' ) || 'ubuntu-latest' }}
+    needs: [check-feature-freeze-event, prepare-for-code-freeze, run-code-freeze, bump-version-in-trunk, build-dev-release, publish-dev-release, notify-slack, trigger-webhook]
+    if: always() && contains(needs.*.result, 'failure')
```

Comment:
Should also check the `cancelled` status similar to WP core https://github.com/WordPress/wordpress-develop/blob/trunk/.github/workflows/phpunit-tests.yml#L325C71-L325C80